src/SearchService.ts

```
@@ -0,0 +1,49 @@
+
+  async runSearch(
+    query: string, 
+    // تحديد القيم الافتراضية بدقة لمنع خطأ الـ {}
+    options: AdvancedSearchOptions = { lemma: false, root: false }, 
+    pagination: PaginationOptions = { page: 1, limit: 10 }
+  ): Promise<any> {
+    
+    // حل مشكلة "Object is possibly null" باستخدام التحقق الشرطي
+    if (this.worker) {
+      const currentWorker = this.worker; // تثبيت المتغير محلياً للـ TypeScript
+      
```

ESLint

[error] Delete `······`

(prettier/prettier)